### ISSUE TITLE
Rule "no-unused-prop-types" does not skip shape props properly

### ISSUE THREAD

**Issue by tristanbbq** [yannickcr/eslint-plugin-react#837]
Rule:
`"react/no-unused-prop-types": ["warn", { "customValidators": [], "skipShapeProps": true }],`

Code sample:
```
import React, { PropTypes } from 'react';
import {
  ScrollView,
  Text,
} from 'react-native';

const UserDetails = ({ user }) => (
  <ScrollView>
    <Text>{user.name}</Text>
    <Text>{user.age}</Text>
    <Text>{user.gender}</Text>
    <Text>{user.email}</Text>
  </ScrollView>
);

UserDetails.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
    gender: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
};
```
This will give me false positives `PropType is defined but props is never used`.

References:
@EvNaverniouk : https://github.com/yannickcr/eslint-plugin-react/pull/611#issuecomment-247438428

**Comment by EvNaverniouk** [comment]
I'm unable to reproduce this on the latest `6.3.0` version. Is this still happening for you @tristanbbq?

**Comment by tristanbbq** [comment]
@EvNaverniouk sorry didn't get the chance to try it out again yet. Feel free to close if you think it's fixed.
